src/state/vidosResolver.ts

```diff
@@ -0,0 +1,131 @@
+import { Id } from '@iden3/js-iden3-core';
+import { type IStateResolver, type ResolvedState, isGenesisStateId } from './resolver';
+
+type DidResolutionResult = {
+  didResolutionMetadata: unknown;
+  didDocumentMetadata: unknown;
+  didDocument: {
+    id: string;
+    alsoKnownAs: string[];
+    controller: string;
+    verificationMethod: {
+      id: string;
+      type: string;
+      controller: string;
+      stateContractAddress: string;
+      published: boolean;
+      info: {
+        id: string;
+        state: string;
+        replacedByState: string;
+        createdAtTimestamp: string;
+        replacedAtTimestamp: string;
+        createdAtBlock: string;
+        replacedAtBlock: string;
+      };
+      global: {
+        root: string;
+        replacedByRoot: string;
+        createdAtTimestamp: string;
+        replacedAtTimestamp: string;
+        createdAtBlock: string;
+        replacedAtBlock: string;
+      };
+    }[];
+  };
+};
+
+/**
+ * Implementation of {@link IStateResolver} that uses Vidos resolver service to resolve states.
+ * It can serve as drop-in replacement for EthStateResolver.
+ */
+export default class VidosResolver implements IStateResolver {
+  constructor(private readonly resolverUrl: string, private readonly apiKey: string) {}
+
+  // Note: implementation closely resembles EthStateResolver because Vidos resolver internally uses the same contract.
+  // The only difference is the usage of regular HTTP requests instead of web3 calls.
+
+  async rootResolve(state: bigint): Promise<ResolvedState> {
+    const stateHex = state.toString(16);
+
+    const zeroAddress = '11111111111111111111'; // 1 is 0 in base58
+    const did = `did:polygonid:polygon:amoy:${zeroAddress}?gist=${stateHex}`;
+
+    const response = await fetch(`${this.resolverUrl}/${encodeURIComponent(did)}`, {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${this.apiKey}`
+      }
+    });
+    const result = (await response.json()) as DidResolutionResult;
+
+    const globalInfo = result.didDocument.verificationMethod[0].global;
```

Comment:
suggest checking for an error in `result.didResolutionMetadata.error` before going too far as otherwise likely will give an incorrect result